examples/checkout.c

```diff
@@ -265,12 +266,29 @@ int lg2_checkout(git_repository *repo, int argc, char **argv)
 
 	if (match_arg_separator(&args)) {
 		/**
-		 * Try to checkout the given path
+		 * Try to checkout the given path(s)
 		 */
 
-		fprintf(stderr, "unhandled path-based checkout\n");
-		err = 1;
-		goto cleanup;
+		git_checkout_options copts = GIT_CHECKOUT_OPTIONS_INIT;
+		git_strarray paths;
+
+		copts.checkout_strategy = GIT_CHECKOUT_FORCE;
```

Comment:
Sorry that I'm getting to this quite late.  Why are we upgrading this to a force checkout when paths are specified?  These should be orthogonal.

Reply:
The purpose here is to be able to do the same as `git checkout -- path/to/file.txt` which will cause changes in that file to be reverted.

If I don't upgrade to force, then changes in the given path are not reverted.

Not how sure how to do this without force.

Reply:
then that's a bug somewhere - either the example is misusing the api elsewhere or `git_checkout` isn't working right in this case.  what is the state of the files in the working directory and index before you run this?  are there staged changes?  does the working directory match the index or head or neither?

Reply:
Below is the code from my test case. As you can see it creates a clean repository, writes a file that is added and committed. Then make some modifications and calling the checkout method to discard the changes.

```
        FS.mkdir('test');
        FS.chdir('test');
        lg.callMain(['init', '.']);

        FS.writeFile('test.txt', 'abcdef');
        lg.callMain(['add', 'test.txt']);
        lg.callMain(['commit', '-m', 'test commit']);
        assert.equal(FS.readFile('test.txt', {encoding: 'utf8'}), 'abcdef');
        FS.writeFile('test.txt', 'abcdefg');
        lg.callMain(['checkout', '--', 'test.txt']);
        lg.callMain(['status']);
        assert.equal(FS.readFile('test.txt', {encoding: 'utf8'}), 'abcdef',
                'expecting file content to be reverted');
```

from here: https://github.com/petersalomonsen/wasm-git/blob/master/test/checkout.spec.js#L21

Reply:
Your test should work as-is, without setting the force option.  Just like it works in the git command-line client.  (`['checkout', '--', 'test.txt']` shouldn't imply `--force`.)  This is a bug that we should fix, whether it's in the example or `git_checkout_head`.

Reply:
all right. I'll see if I can find some time to investigate it :-) would be good to sort it out.

Reply:
Just to be clear - I wasn't suggesting that you needed to be the one to do this work.  :)  Just that we should figure out what's going on here.  The examples should show new users the best practices for using the APIs and we don't necessarily want people defaulting to force when it's not (or - at least - shouldn't be) necessary.

Reply:
sure, I didn't get the impression that you suggested I should do it :-) but then it's annoying too, so can't help not trying to figure out what it is :-)

I found something here:
https://github.com/libgit2/libgit2/blob/master/src/checkout.c#L499

changing from:
`*action = CHECKOUT_ACTION_IF(FORCE, UPDATE_BLOB, NONE);`

to

`*action = CHECKOUT_ACTION__UPDATE_BLOB;`

fixes so that I don't have to use `FORCE`.

However a test should be made for this to be sure that this check does not serve other purposes.

Reply:
ok I found a test that relies on this (checking out without given paths should not overwrite modified files by default), so I wrote a bugfix so that this test pass, and also added a new test for checking out with given paths.

does it look ok? @ethomson @pks-t 

if so, would you like me to add an issue, and reference this in this PR (and the commit comment)?